perun-engine/src/main/java/cz/metacentrum/perun/engine/scheduling/impl/SendWorkerImpl.java

```diff
@@ -92,14 +94,20 @@ public SendTask call() throws TaskExecutionException {
 						task.getId(), getReturnCode(), getStdout(), getStderr());
 
 				sendTask.setStatus(ERROR);
+				// XXX: why exception? There is nothing exceptional about the situation.
```

Comment:
I guess its historically by the way the Collectors are written. By their expectations, if Task succeeds, then Task/SendTask is returned. When it fails Exception is thrown and it contains all relevant error codes and messages, since in case of Task, there is no property for them to be passed. Like this collector has basically one positive and one negative branch, instead of three.